backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/work/builder/factory/impl/ElasticsearchWorkBuilderFactory.java

```diff
@@ -0,0 +1,78 @@
+import org.hibernate.search.backend.elasticsearch.work.builder.impl.CreateIndexWorkBuilder;
+import org.hibernate.search.backend.elasticsearch.work.builder.impl.DeleteByQueryWorkBuilder;
+import org.hibernate.search.backend.elasticsearch.work.builder.impl.DeleteWorkBuilder;
+import org.hibernate.search.backend.elasticsearch.work.builder.impl.DropIndexWorkBuilder;
+import org.hibernate.search.backend.elasticsearch.work.builder.impl.ExplainWorkBuilder;
+import org.hibernate.search.backend.elasticsearch.work.builder.impl.FlushWorkBuilder;
+import org.hibernate.search.backend.elasticsearch.work.builder.impl.GetIndexSettingsWorkBuilder;
+import org.hibernate.search.backend.elasticsearch.work.builder.impl.IndexExistsWorkBuilder;
+import org.hibernate.search.backend.elasticsearch.work.builder.impl.IndexWorkBuilder;
+import org.hibernate.search.backend.elasticsearch.work.builder.impl.OpenIndexWorkBuilder;
+import org.hibernate.search.backend.elasticsearch.work.builder.impl.OptimizeWorkBuilder;
+import org.hibernate.search.backend.elasticsearch.work.builder.impl.PutIndexSettingsWorkBuilder;
+import org.hibernate.search.backend.elasticsearch.work.builder.impl.RefreshWorkBuilder;
+import org.hibernate.search.backend.elasticsearch.work.builder.impl.ScrollWorkBuilder;
+import org.hibernate.search.backend.elasticsearch.work.builder.impl.SearchWorkBuilder;
+
+import com.google.gson.JsonObject;
+
+/**
+ * @author Yoann Rodiere
+ */
+public interface ElasticsearchWorkBuilderFactory {
+
+	IndexWorkBuilder index(URLEncodedString indexName, URLEncodedString typeName, URLEncodedString id, JsonObject document);
+
+	DeleteWorkBuilder delete(URLEncodedString indexName, URLEncodedString typeName, URLEncodedString id);
+
+	DeleteByQueryWorkBuilder deleteByQuery(URLEncodedString indexName, JsonObject payload);
+
+	FlushWorkBuilder flush();
+
+	RefreshWorkBuilder refresh();
+
+	OptimizeWorkBuilder optimize();
+
+	// TODO restore method => BulkWorkBuilder bulk(List<BulkableElasticsearchWork<?>> bulkableWorks);
+
+	SearchWorkBuilder search(JsonObject payload);
+
+	ExplainWorkBuilder explain(URLEncodedString indexName, URLEncodedString typeName, URLEncodedString id, JsonObject payload);
+
+	ScrollWorkBuilder scroll(String scrollId, String scrollTimeout);
+
+	ClearScrollWorkBuilder clearScroll(String scrollId);
+
+	CreateIndexWorkBuilder createIndex(URLEncodedString indexName);
+
+	DropIndexWorkBuilder dropIndex(URLEncodedString indexName);
+
+	OpenIndexWorkBuilder openIndex(URLEncodedString indexName);
+
+	CloseIndexWorkBuilder closeIndex(URLEncodedString indexName);
+
+	IndexExistsWorkBuilder indexExists(URLEncodedString indexName);
+
+	GetIndexSettingsWorkBuilder getIndexSettings(URLEncodedString indexName);
+
+	PutIndexSettingsWorkBuilder putIndexSettings(URLEncodedString indexName, IndexSettings settings);
+
+	// TODO restore method => GetIndexTypeMappingsWorkBuilder getIndexTypeMappings(URLEncodedString indexName);
```

Comment:
Is there something similar to `org.hibernate.search.elasticsearch.schema.impl.model.TypeMapping` in v6?

Reply:
Right, @gsmet renamed that because we both thought it was cleaner for various reasons. 

`TypeMapping` has been split in two: `AbstractTypeMapping`, the base class, and `RootTypeMapping`, the concrete class.

In this case I think you need `RootTypeMapping`. That's the one you used in `org.hibernate.search.backend.elasticsearch.work.impl.CreateIndexWork.Builder#mapping`.